waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -972,6 +1013,32 @@
   (get-services [this]
     (get-services this))
 
+  (signal-instance [this {:keys [id service-id] :as instance} signal-type]
+    (ss/try+
+      (let [service (service-id->service this service-id)
+            {:keys [pod-name]} (unpack-instance-id id)
+            pod (get-in @watch-state [:service-id->pod-id->pod service-id pod-name])
+            service-instance (pod->ServiceInstance this pod)
+            response (signal-service-instance this service-instance service signal-type)]
```

Comment:
We should check if the service-id and instance-id are valid before invoking this function and reporting appropriate responses.